src/test/java/com/google/sps/CapstoneAuthTest.java

```diff
@@ -0,0 +1,98 @@
+package com.google.sps;
+
+import java.io.IOException;
+
+
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import static org.junit.Assert.*;
+
+import static org.mockito.Mockito.*;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import com.google.cloud.datastore.Key;
+
+import com.google.sps.CapstoneAuth;
+import static com.google.sps.ChatWebSocket.*;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseToken;
+import com.google.firebase.auth.UserRecord;
+import com.google.firebase.auth.FirebaseAuthException;
+
+
+@PrepareForTest({CapstoneAuth.class, UserRecord.class, FirebaseAuth.class,
+    FirebaseToken.class})
+@RunWith(PowerMockRunner.class)
+public class CapstoneAuthTest {
+
+    public static final String TEST_ID_TOKEN = "1234testtoken";
+    public static final String TEST_UID = "testuid";
+    public static final String TEST_FALSE_UID = "testfalseuid";
+    public static final String TEST_EMAIL = "test@email.com";
+
+    private CapstoneAuth obj;
+
+    private FirebaseToken mockToken;
+    private FirebaseAuth mockFirebase;
+    private UserRecord mockUser; 
+
+    @Before
+    public void setUp() throws NoSuchMethodException, InstantiationException, IllegalAccessException, FirebaseAuthException {
+
+        mockFirebase = mock(FirebaseAuth.class);
+        PowerMockito.mockStatic(FirebaseAuth.class);
+
+        mockUser = mock(UserRecord.class);
+        PowerMockito.mockStatic(UserRecord.class);
+
+        mockToken = mock(FirebaseToken.class);
+        PowerMockito.mockStatic(FirebaseToken.class);
+
+        when(FirebaseAuth.getInstance()).thenReturn(mockFirebase);
+
+        when(mockFirebase.getUser(TEST_UID)).thenReturn(mockUser);
+        when(mockUser.getEmail()).thenReturn(TEST_EMAIL);
+
+        when(mockFirebase.verifyIdToken(TEST_UID)).thenReturn(mockToken);
+        when(mockToken.getUid()).thenReturn(TEST_ID_TOKEN);
+
+        obj = (CapstoneAuth) CapstoneAuth.getInstance();
```

Comment:
Rather than calling getInstance to retrieve a new CapstoneAuth object, I would recommend using the java.lang.reflect library to make the private constructor accessible and then use that to create a new instance of the class. That way, if getInstance() does not work as intended, you can ensure the obj will be initialized correctly.

Reply:
That's a good point. The issue I ran into was "java.lang.IllegalStateException: FirebaseApp name [DEFAULT] already exists!". My guess is that something with the java.lang.reflect library makes Firebase break the singleton class pattern and my solution, tho imperfect, preserves the singleton class pattern. I tried the the java.lang.constructor route yesterday to no avail, but if you (or anybody) has other ideas, I'm all ears.